src/main/java/com/ericsson/ei/frontend/utils/BackEndInformationControllerUtils.java

```diff
@@ -156,8 +157,9 @@
                         getHeaders(), HttpStatus.BAD_REQUEST);
             }
         } catch (Exception e) {
+            LOG.error(ExceptionUtils.getStackTrace(e));
             return new ResponseEntity<>(
-                    "Internal error, " + e.getMessage(),
+                    "{\"message\": \"Internal server error.\"}",
```

Comment:
Same comment here as well,, give some more info about what the issue is back to the user.

Reply:
The thing is, this issue is not relevant to the browser as nothing should have been wrong with the contents from the browser, the other error messages is for that. If it comes to the point where we have an internal server error it should be enough to say that there was an internal server error if the problem is some sort of unhandled issue in the webserver. If it ever comes to the point where this happens, there's a bug that we should fix anyways and then they have the log to report this with.

Reply:
I agree with Erik here, the e-messages is for admins and not users. In a big applications all errors would be handled and given an error code that the user could report further.

Reply:
When it comes to the internal server error, the user cannot not solve it without access to the deployed system. It is a bug or environmental issues in the back-end as it is an "internal server error". Therefore it's not something that the user have done wrong, it's an error in the server itself. If there however is anything wrong with the request the user sends, the user should get an answer of what was wrong in their request, and this is done in the code. 

This also follows some standard REST API error handling best practices as mentioned in https://blog.restcase.com/rest-api-error-codes-101/
4xx: Client Error - This category of error status codes points the finger at clients.
5xx: Server Error - The server takes responsibility for these error status codes.

Also, sending stacktrace information back to the user in a REST system can also result in exposing data/structure from for example the database or other "secret" information in the deployed system. 

If we send something back to the browser about an internal error i think it should be done as Anders mentioned, send an error code and document the error codes.

Reply:
Often an exception has a message with it so use that message in the ResponseEnity if it exist, otherwise use the one you use now.